### PR TITLE
Fix __traits(isDeprecated) for complex types

### PR DESCRIPTION
**Pull request by wilzbach**
Introduced in https://github.com/dlang/dmd/pull/7665

### CODE REVIEW

File: src/dmd/traits.d [hunk 1]
```diff
@@ -504,7 +504,7 @@ extern (C++) Expression semanticTraits(TraitsExp e, Scope* sc)
     {
         if (global.params.vcomplex)
         {
-            if (isTypeX(t => t.iscomplex() || t.isimaginary()))
+            if (isTypeX(t => t.iscomplex() || t.isimaginary()).value == 1)
```
Comment (wilzbach) [inline src/dmd/traits.d]: `isTypeX` returns a class instance (now `IntegerExp`) which will always be `true` ...

Comment (JinShil) [reply src/dmd/traits.d]: Gah!

**Review by JinShil — APPROVED** [review]

File: src/dmd/traits.d (hunk 1 above)
Comment (ibuclaw) [reply src/dmd/traits.d]: And you couldn't have used `isTypeX(...).isBool(true)`?

Comment (wilzbach) [reply src/dmd/traits.d]: -> https://github.com/dlang/dmd/pull/7722

**Pull request merged**